src/AutoMapper/XpressionMapper/XpressionMapperVisitor.cs

```diff
@@ -84,20 +87,43 @@ protected override Expression VisitMember(MemberExpression node)
                     ? visitor.Visit(last.CustomExpression.Body.MemberAccesses(afterCustExpression))
                     : visitor.Visit(last.CustomExpression.Body);
 
-                this.TypeMappings.AddTypeMapping(node.Type, ex.Type);
+                this.TypeMappings.AddTypeMapping(ConfigurationProvider, node.Type, ex.Type);
                 return ex;
             }
             fullName = BuildFullName(propertyMapInfoList);
             var me = ExpressionFactory.MemberAccesses(fullName, InfoDictionary[parameterExpression].NewParameter);
 
-            this.TypeMappings.AddTypeMapping(node.Type, me.Type);
+            this.TypeMappings.AddTypeMapping(ConfigurationProvider, node.Type, me.Type);
             return me;
+        }
+
+        protected override Expression VisitUnary(UnaryExpression node)
+        {
+            switch (node.NodeType)
+            {
+                case ExpressionType.Convert:
+                case ExpressionType.ConvertChecked:
+                    switch (node.Operand.NodeType)
+                    {
+                        case ExpressionType.Constant:
+                            return ProcessConstant(node.Operand as ConstantExpression);
```

Comment:
A simple cast instead of as. But wouldn't VisitConstant below be called anyway? I'm guessing not :), but it's kind of weird.

Reply:
Yep - (no VisitConstant does not get called) -  if you know another way ...